db.php

```diff
@@ -91,25 +91,55 @@ public function get_all_entries($userid, $mode = null) {
         // Select the website, username, password, and ID for all password entries
         // that belong to the user and meet the specified filtering criteria
 
-        if ($mode === 'favorite') {
-            $favoriteSelection = true;
-        } elseif ($mode === 'trash') {
-            $favoriteSelection = null;
-        } else {
-            $favoriteSelection = Medoo::raw('favorite IS NULL OR favorite = true');
+        switch ($mode) {
+            case 'trash':
+                $results = $this->database->select('passwords', [
+                    'website',
+                    'username',
+                    'password',
+                    'id'
+                ], [
+                    'user_id' => $userid,
+                    'trash' => true,
+                    'OR' => [
+                        'favorite' => true,
+                        'favorite' => null,
+                        'trash' => true,
```

Comment:
Isn't that trash condition redundant because it is already included in the main condition `'trash' => true`?

Reply:
You would think so, but when you try to remove it, there are not all passwords displayed.

Reply:
Ok weird.
But it as long as the code works we can just keep it in our minds and investigate this issue later.